sumpy/p2p.py

```diff
@@ -459,19 +459,42 @@ def get_kernel(self):
                     shape="nstrengths, nsources", dim_tags="sep,C"),
                 lp.GlobalArg("result", None,
                     shape="noutputs, ntargets", dim_tags="sep,C"),
+                lp.TemporaryVariable("tgt_center", shape=(self.dim,)),
                 "..."
-            ])
+            ]
 
-        loopy_knl = lp.make_kernel([
+        domains = [
             "{[itgt_box]: 0 <= itgt_box < ntgt_boxes}",
+            "{[iknl]: 0 <= iknl < noutputs}",
             "{[isrc_box]: isrc_box_start <= isrc_box < isrc_box_end}",
-            "{[itgt, isrc, idim]: \
-                itgt_start <= itgt < itgt_end and \
-                isrc_start <= isrc < isrc_end and \
-                0 <= idim < dim}",
-            ],
-            self.get_kernel_scaling_assignments()
-            + ["""
+            "{[idim]: 0 <= idim < dim}",
+            "{[istrength]: 0 <= istrength < nstrengths}",
+            "{[isrc]: isrc_start <= isrc < isrc_end}"
+        ]
+
+        src_outer_limit = (max_nsources_in_one_box - 1) // nsplit
+        tgt_outer_limit = (max_ntargets_in_one_box - 1) // nsplit
+
+        if gpu:
```

Comment:
If you run the GPU kernel on the CPU, does it do worse? (What I'm aiming for is to only have one of them.)

Reply:
Pocl miscompiles the GPU kernel and there are NaNs as output in some tests.

Reply:
* Only pocl-cpu or all pocls?
* Could you add a comment to explain why we have both?

Reply:
pocl-cpu only. Added a comment in 955d279f68daa4d5103b1165ef8c70f075b9dd6f